Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp

```diff
@@ -705,10 +721,17 @@ void pfPatcher::OnSelfPatch(FileDownloadFunc cb)
 
 // ===================================================
 
-void pfPatcher::RequestGameCode()
+void pfPatcher::RequestGameCode(bool python, bool sdl)
 {
+    fWorker->fWantPython = python;
+    fWorker->fWantSDL = sdl;
+
     hsLockGuard(fWorker->fRequestMut);
-    fWorker->fRequests.emplace_back("SecurePreloader", pfPatcherWorker::Request::kSecurePreloader);
+    if (NetCliFileQueryConnected()) {
```

Comment:
All the other code checks `gDataServerLocal` directly - or is this condition supposed to check something different?

Reply:
`gDataServer` is a variable from `plResManager`, which I really didn't want to depend on in this case. Instead, what we're really interested in is whether or not a file server is available. If it isn't. then we should fall back to the auth server.

Reply:
> what we're really interested in is whether or not a file server is available. If it isn't. then we should fall back to the auth server.

Ah, I see - that logic makes sense, and in that case I agree the new function is the better choice.